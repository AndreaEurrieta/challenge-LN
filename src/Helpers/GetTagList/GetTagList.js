export const GetTagList = (items = []) => {
    const filterTags = (items.map(article => article.taxonomy.tags)).flat()

    const orderedArray = []

    const countTags = filterTags.map(article => article.slug)
        .reduce((obj, slug) => {
            if (obj[slug]) {
                obj[slug] = obj[slug] + 1;
            } else {
                obj[slug] = 1;
            }
            return obj;
        }, {})

    for (const slugObj in countTags) {
        const { text } = filterTags.find(({ slug }) => slug === slugObj)
        orderedArray.push({
            slug: slugObj,
            text: text,
            count: countTags[slugObj]
        })
    }

    orderedArray.sort((a, b) => a.count - b.count).reverse()

    const newArray = orderedArray.slice(0, 10)

    return newArray
}
