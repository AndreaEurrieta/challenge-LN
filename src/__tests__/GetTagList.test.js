import { GetTagList } from '../Helpers/GetTagList/GetTagList';
import tagMock from '../__mocks__/tagMock.json' 

describe("Helpers-GetTagList", () => {
    test("Debería devolver diez objetos", () => {
      const tags = GetTagList(tagMock);
      expect(tags).toHaveLength(10);
      expect(tags).toMatchSnapshot();
    });
    test("Debería devolver true si no recibe ningún parametro", () => {
      expect(GetTagList()).toBeTruthy();
    });
  });