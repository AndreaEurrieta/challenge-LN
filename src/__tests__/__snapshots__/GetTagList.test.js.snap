// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers-GetTagList Debería devolver diez objetos 1`] = `
Array [
  Object {
    "count": 13,
    "slug": "huevo-tid47236",
    "text": "Huevo",
  },
  Object {
    "count": 9,
    "slug": "leche-tid47244",
    "text": "Leche",
  },
  Object {
    "count": 7,
    "slug": "azucar-tid47141",
    "text": "Azúcar",
  },
  Object {
    "count": 7,
    "slug": "manteca-tid47257",
    "text": "Manteca",
  },
  Object {
    "count": 7,
    "slug": "arroz-tid47136",
    "text": "Arroz",
  },
  Object {
    "count": 5,
    "slug": "harina-0000-tid48184",
    "text": "Harina 0000",
  },
  Object {
    "count": 4,
    "slug": "frutas-tid67217",
    "text": "Frutas",
  },
  Object {
    "count": 4,
    "slug": "ajo-tid47126",
    "text": "Ajo",
  },
  Object {
    "count": 4,
    "slug": "pescados-tid67216",
    "text": "Pescados",
  },
  Object {
    "count": 3,
    "slug": "cebolla-morada-tid47808",
    "text": "Cebolla morada",
  },
]
`;
